homeassistant/components/tradfri/base_class.py

```diff
@@ -0,0 +1,96 @@
+"""Base class for IKEA TRADFRI."""
+import logging
+
+from pytradfri.error import PytradfriError
+
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from . import DOMAIN as TRADFRI_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TradfriBaseDevice(Entity):
+    """Base class for a TRADFRI device."""
+
+    def __init__(self, device, api, gateway_id):
+        """Initialize a device."""
+        self._available = True
+        self._api = api
+        self._device = None
+        self._device_control = None
+        self._device_data = None
+        self._gateway_id = gateway_id
+        self._name = None
+        self._unique_id = None
+
+        self._refresh(device)
+
+    @callback
+    def _async_start_observe(self, exc=None):
+        """Start observation of device."""
+        if exc:
+            self._available = False
+            self.async_schedule_update_ha_state()
+            _LOGGER.warning("Observation failed for %s", self._name, exc_info=exc)
+
```

Comment:
Make sure to return or you will still execute the other code ?

```suggestion
           return
```

Reply:
And if observation did fail, should you retry right away? Should we have some back off.

Reply:
I'm afraid I'm not very familiar with how this part of the code works (looks like you and Lwis wrote it a long time ago.) I know as much that the `_async_start_observe` looks more or less the same in all of the Tradfri device types, so we should be able to move it into the new base class.

However, I wouldn't know what I'm doing when making changes to `_async_start_observe`. It would be a lot of guessing on my end. I'd be happy to just put in whatever you suggests. But that will require time for you to figure out the code. Perhaps a better idea to leave this method as is in the switch-file?